### PR TITLE
Payments/enterprise

### PR DESCRIPTION
**Pull request by motechFR**
### <samp>🤖 Generated by Copilot at 55b40db</samp>

This pull request adds support for the enterprise paid tier in the subscription settings and the stripe integration. It introduces a new component `EnterpriseBillingScreen` to display the plan and billing details, and several utility functions to handle the stripe customer metadata and updates. It also improves the error handling and logging in the API modules.

### WHY
Add Enterprise workflow

### CODE REVIEW

File: pages/api/v1/webhooks/stripe/index.ts [hunk 1]
```diff
@@ -107,8 +116,8 @@ export async function stripePayment(req: NextApiRequest, res: NextApiResponse):
           deletedAt: null
         };
 
-        await prisma.$transaction([
-          prisma.stripeSubscription.upsert({
+        await prisma.$transaction(async (tx) => {
+          await tx.stripeSubscription.upsert({
```
Comment (mattcasey) [inline pages/api/v1/webhooks/stripe/index.ts:120]: passing in tx doesn't change the original behavior right?

**Review by mattcasey — APPROVED** [review]

File: pages/api/v1/webhooks/stripe/index.ts (hunk 1 above)
Comment (motechFR) [reply pages/api/v1/webhooks/stripe/index.ts]: Yep, was going to make a different change, but reverted back to original.

I've added better fallback handling for the paidTier setter

File: scripts/migrations/2023_27_06_free-trials.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  for (let i = 0; i < totalSpaces; i++) {
+    const space = spacesToProvision[i]
+    await rateLimit();
+    await createProSubscription({
+      //    billingEmail: undefined as any,
+      period: 'monthly',
+      spaceId: space.id,
+      name: space.name,
+      freeTrial: true,
+      blockQuota: 30
```
Comment (mattcasey) [inline scripts/migrations/2023_27_06_free-trials.ts:35]: @motechFR isn't this supposed to be 90 days?

File: public/images/paidtiers/enterprise.png
Comment (mattcasey) [inline public/images/paidtiers/enterprise.png:1]: this picture probably shouldn't have a black border on it... is it the only one we had?

**Pull request merged**